app/V1Module/presenters/AssignmentSolutionReviewsPresenter.php

```diff
@@ -0,0 +1,407 @@
+ * @LoggedIn
+ */
+class AssignmentSolutionReviewsPresenter extends BasePresenter
+{
+    /**
+     * @var ReviewComments
+     * @inject
+     */
+    public $reviewComments;
+
+    /**
+     * @var AssignmentSolutions
+     * @inject
+     */
+    public $assignmentSolutions;
+
+    /**
+     * @var Users
+     * @inject
+     */
+    public $users;
+
+    /**
+     * @var IAssignmentSolutionPermissions
+     * @inject
+     */
+    public $assignmentSolutionAcl;
+
+    /**
+     * @var IUserPermissions
+     * @inject
+     */
+    public $userAcl;
+
+    /**
+     * @var AssignmentSolutionViewFactory
+     * @inject
+     */
+    public $assignmentSolutionViewFactory;
+
+    /**
+     * @var ReviewsEmailsSender
+     * @inject
+     */
+    public $reviewsEmailSender;
+
+
+    public function checkDefault(string $id)
+    {
+        $solution = $this->assignmentSolutions->findOrThrow($id);
+        if (!$this->assignmentSolutionAcl->canViewReview($solution)) {
+            throw new ForbiddenRequestException("You cannot access review of this solution");
+        }
+    }
+
+    /**
+     * Get detail of the solution and a list of review comments.
+     * @GET
+     * @param string $id identifier of the solution
+     * @throws InternalServerException
+     */
+    public function actionDefault(string $id)
+    {
+        $solution = $this->assignmentSolutions->findOrThrow($id);
+        $this->sendSuccessResponse([
+            "solution" => $this->assignmentSolutionViewFactory->getSolutionData($solution),
+            "reviewComments" => $solution->getReviewComments()->toArray(),
+        ]);
+    }
+
+    public function checkUpdate(string $id)
+    {
+        $solution = $this->assignmentSolutions->findOrThrow($id);
+        if (!$this->assignmentSolutionAcl->canReview($solution)) {
+            throw new ForbiddenRequestException("You cannot perform a review of this solution");
+        }
+    }
+
+    /**
+     * Update the state of the review process of the solution.
+     * @POST
+     * @Param(type="post", name="close", validation="bool"
+     *        description="If true, the review is closed. If false, the review is (re)opened.")
+     * @param string $id identifier of the solution
+     * @throws InternalServerException
+     */
+    public function actionUpdate(string $id)
+    {
+        $solution = $this->assignmentSolutions->findOrThrow($id);
+        $close = filter_var($this->getRequest()->getPost("close"), FILTER_VALIDATE_BOOLEAN);
+        $reviewedAt = $solution->getReviewedAt();
+
+        if ($solution->getReviewStartedAt() === null) {
+            $solution->setReviewStartedAt(new DateTime()); // the review is marked as opened in any case
+        }
+
+        if ($close && $reviewedAt === null) {
+            // opened -> closed transition
+            $solution->setReviewedAt(new DateTime());
+
+            $issues = 0; // issues are duly counter only when the review is closed
```

Comment:
Maybe add this comment also to the AssignmentSolutionEntity->issuesCount member?